### PR TITLE
Addition of a file change detection mechanism.

### PR DESCRIPTION
**Pull request by ZePan110**
## Description

The summary of the proposed changes as long as the relevant motivation and context.

## Issues

List the issue or RFC link this PR is working on. If there is no such link, please mark it as `n/a`.

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would break existing design and interface)

## Dependencies

List the newly introduced 3rd party dependency if exists.

## Tests

Describe the tests that you ran to verify your changes.

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Check for changed Dockerfile paths
+        run: |
+          merged_commit=$(git log -1 --format='%H')
```
Comment (mkbhanda) [inline .github/workflows/test.yaml:27]: is hours granularity adequate?

**Review by chensuyue — APPROVED** [review]

File: .github/workflows/test.yaml (hunk 1 above)
Comment (chensuyue) [reply .github/workflows/test.yaml:27]: `-1`: This option limits the output to just the most recent commit.
`--format='%H'`: This specifies the format of the output. %H is a format specifier that outputs the full commit hash.

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+                used_files+="$file "
+              fi
+            done
+          fi
+
+          if [[ "$is_use" == "TRUE" ]]; then
+            echo "Warning: Changed Dockerfile paths:"
+            echo "$used_files"
+            echo "Please modify the corresponding README in GenAIExamples repo and ask suyue.chen@intel.com for final confirmation."
+            exit 1
```
Comment (daisy-ycguo) [inline .github/workflows/test.yaml:54]: I guess this job is a warning. Returning 1 will cause it failure. Maybe remove the error return statement and just return correctly.
What's more, if only the content of docker file is changed and not the file path, does the developer need to modify READMEs in GenAIExamples?
I think maybe the logic is like: find the removed (renamed) docker files in the changes log, and check if the original docker file path is used in READMEs in GenAIExamples.

**Review by daisy-ycguo — APPROVED** [review]
LGTM. Thank you.

**Review by kevinintel — APPROVED** [review]

**Pull request merged**